python/whylogs/api/store/profile_store.py

```diff
@@ -1,51 +1,28 @@
 from abc import ABC, abstractmethod
-from datetime import datetime
-from typing import Any, Dict, List, Optional
+from datetime import timedelta
+from typing import List, Optional
 
-from whylogs.api.writer.writer import Writable
-from whylogs.core import DatasetProfile, Segment
-from whylogs.core.schema import DatasetSchema
-from whylogs.core.stubs import pd
+from whylogs.api.store.query import DateQuery
+from whylogs.core import DatasetProfileView
 
 
 class ProfileStore(ABC):
-    """
-    A holder object for profiles.
-    """
-
-    @abstractmethod
-    def get_matching_profile(
-        self,
-        obj: Any = None,
-        *,
-        pandas: Optional[pd.DataFrame] = None,
-        row: Optional[Dict[str, Any]] = None,
-        segment: Optional[Segment] = None,
-        schema: Optional[DatasetSchema] = None,
-    ) -> List[DatasetProfile]:
-        return [DatasetProfile(schema)]
-
     @abstractmethod
-    def read(
-        self,
-        id: Optional[str],
-        dataset_timestamp: Optional[datetime] = None,
-        segment: Optional[Segment] = None,
-        schema: Optional[DatasetSchema] = None,
-    ) -> Optional[DatasetProfile]:
-        return [DatasetProfile(schema)]
-
-    @abstractmethod
-    def write(
-        self,
-        data: Writable,
-        model_id: Optional[str] = None,
-        dataset_timestamp: Optional[datetime] = None,
-        segment: Optional[Segment] = None,
-        schema: Optional[DatasetSchema] = None,
-    ) -> None:
+    def get(self, query: DateQuery) -> DatasetProfileView:
         pass
```

Comment:
Just occurred to me, but I guess we don't have to worry too much about having a common base query type, we can just use `Union[DateQuery, ProfileIdQuery, etc]`

Reply:
That lets us handle each one differently internally without shoehorning them all into returning some common type

Reply:
sounds good! i will resolve this for now and when we have a more mature idea on a `ProfileIdQuery` class we will add an `Union` type there ;)